src/jit/decomposelongs.cpp

```diff
@@ -1642,49 +1642,83 @@ GenTree* DecomposeLongs::DecomposeSimdGetItem(LIR::Use& use)
     assert(varTypeIsSIMD(simdTree->gtOp.gtOp1->gtType));
     assert(simdTree->gtOp.gtOp2->gtType == TYP_INT);
 
+    bool    indexIsConst = simdTree->gtOp.gtOp2->IsCnsIntOrI();
+    ssize_t index        = 0;
+    if (indexIsConst)
+    {
+        index = simdTree->gtOp.gtOp2->gtIntCon.gtIconVal;
+    }
+
     LIR::Use op1(Range(), &simdTree->gtOp.gtOp1, simdTree);
     unsigned simdTmpVarNum = op1.ReplaceWithLclVar(m_compiler, m_blockWeight);
     JITDUMP("[DecomposeSimdGetItem]: Saving op1 tree to a temp var:\n");
     DISPTREERANGE(Range(), op1.Def());
 
-    LIR::Use op2(Range(), &simdTree->gtOp.gtOp2, simdTree);
-    unsigned indexTmpVarNum = op2.ReplaceWithLclVar(m_compiler, m_blockWeight);
-    JITDUMP("[DecomposeSimdGetItem]: Saving op2 tree to a temp var:\n");
-    DISPTREERANGE(Range(), op2.Def());
-
-    // TODO-CQ: if the index is constant, we don't need to do the computation dynamically.
+    unsigned indexTmpVarNum = 0;
+    if (!indexIsConst)
+    {
+        LIR::Use op2(Range(), &simdTree->gtOp.gtOp2, simdTree);
+        indexTmpVarNum = op2.ReplaceWithLclVar(m_compiler, m_blockWeight);
+        JITDUMP("[DecomposeSimdGetItem]: Saving op2 tree to a temp var:\n");
+        DISPTREERANGE(Range(), op2.Def());
+    }
 
     // Create:
-    //      loResult = GT_SIMD{get_item}[int](tmp_simd_var, tmp_index * 2)
+    //      loResult = GT_SIMD{get_item}[int](tmp_simd_var, index * 2)
```

Comment:
It would be nice to update this comment since we're not using the tmp_simd_var inthe case where `indexIsConst`.